common/calculate.ts

```diff
@@ -158,13 +158,53 @@ function getDpmInvested(yourBets: Bet[]) {
   })
 }
 
+export function getContractPositions(yourBets: Bet[]) {
+  let loan = 0
+  const totalShares: { [outcome: string]: number } = {}
+
+  for (const bet of yourBets) {
+    const { loanAmount, shares, sharesByOutcome, outcome } = bet
+
+    if (sharesByOutcome) {
+      for (const [o, s] of Object.entries(sharesByOutcome)) {
+        totalShares[o] = (totalShares[o] ?? 0) + s
+      }
+    } else {
+      totalShares[outcome] = (totalShares[outcome] ?? 0) + shares
+    }
+
+    loan += loanAmount ?? 0
+  }
+
+  const hasShares = Object.values(totalShares).some(
+    (shares) => !floatingEqual(shares, 0)
+  )
+
+  const { YES: yesShares, NO: noShares } = totalShares
+  const hasYesShares = yesShares >= 1
+  const hasNoShares = noShares >= 1
+  const lastBetTime = Math.max(...yourBets.map((b) => b.createdTime))
+  const maxSharesOutcome = hasShares
+    ? maxBy(Object.keys(totalShares), (outcome) => totalShares[outcome])
+    : null
+
+  return {
```

Comment:
I think `invested` should be included as part of your position and not as a metric, because it's not affected by other people's bets.